docs/user/reference/hook.md

```diff
@@ -137,6 +142,236 @@ The result is that the events are consumed in the order: `2-1-3`. Beware.
+* `<name>-relation-departed`
+* `<name>-relation-broken`
+
+For each charm relation, any or all of the above relation hooks can be implemented.
+Relation hooks operate in an environment with additional environment variables available:
+
+* JUJU_RELATION is set to the name of the charm relation. This is of limited value, because every relation hook already "knows" what charm relation it was written for; that is, in the `foo-relation-joined` hook, JUJU_RELATION is `foo`.
+* JUJU_RELATION_ID is more useful, because it serves as unique identifier for a particular relation, and thereby allows the charm to handle distinct relations over a single endpoint. In hooks for the `foo` charm relation, JUJU_RELATION_ID always has the form "foo:<id>", where id uniquely but opaquely identifies the runtime relation currently in play.
+* JUJU_REMOTE_APPLICATION is set to the name of the related application.
+
+Furthermore, all relation hooks except relation-created and relation-broken are notifications about some specific unit of a related application, and operate in an environment with the following additional environment variables available:
+
+* JUJU_REMOTE_UNIT is set to the name of the current related unit.
+
+For every relation in which a unit participates, hooks for the appropriate charm relation are run according to the following rules.
+
+The `relation-created` hook always runs once when the relation is first created, before any related units are processed.
+
+The `relation-joined` hook always runs once when a related unit is first seen.
+
+The `relation-changed` hook for a given unit always runs once immediately following the relation-joined hook for that unit, and subsequently whenever the related unit changes its settings (by calling relation-set and exiting without error).
+
+```{note}
+"Immediately" only applies within the context of this particular relation - that is, when
+"foo-relation-joined" is run for unit "bar/99" in relation id "foo:123", the only guarantee is that
+the next hook to be run *in relation id "foo:123"* will be "foo-relation-changed" for "bar/99".
+Non-relation hooks may intervene, as may hooks for other relations, and even for other "foo" relations.
+```
+
+The `relation-departed` hook for a given unit always runs once when a related unit is no longer related. After the `relation-departed` hook has run, no further notifications will be received from that unit; however, its settings will remain accessible via relation-get for the complete lifetime of the relation.
+
+The `relation-broken` hook is not specific to any unit, and always runs once when the local unit is ready to depart the relation itself. Before this hook is run, a relation-departed hook will be executed for every unit known to be related; it will never run while the relation appears to have members, but it may be the first and only hook to run for a given relation. The stop hook will not run while relations remain to be broken.
 
 
 ## List of hooks
```

Comment:
For each hook, regarding the questions: 

I know we've discussed two potential improvements: 

1. Rephrase as "Which environment variables is it executed with?" "What is the context of the hook?"
2. Add another question "What is a charm author expected to do in response to this hook?" 

I think we should leave that for a follow-up PR.